docs/faq.adoc

```diff
@@ -0,0 +1,64 @@
+== FAQs
+
+This document provides a quick reference for frequently questioned concepts and features in Credentials.
+
+=== Where do credentials get stored?
+
+By default, Jenkins uses an internal credentials store where credentials secret data are encrypted at rest in the Jenkins controller file system.
+This security model does not prevent anyone with read access to the controller file system from reading and decrypting those secrets.
+For more sensitive secrets, it is highly recommended to install and use an external credentials provider plugin which delegates the secure storage of secrets to a proper key management system.
+Various cloud providers (including https://plugins.jenkins.io/aws-secrets-manager-credentials-provider/[AWS], https://plugins.jenkins.io/azure-keyvault/[Azure], and https://plugins.jenkins.io/gcp-secrets-manager-credentials-provider/[GCP]) have secrets management and HSMs available for this purpose, and open source software such as https://plugins.jenkins.io/kubernetes-credentials-provider/[Kubernetes], https://plugins.jenkins.io/hashicorp-vault-plugin/[Vault], or https://plugins.jenkins.io/conjur-credentials/[Conjur] provide more generic secrets management for multiple platforms.
```

Comment:
Prefer a list format.

Also, it might be useful to link (with disclaimer about technical content) to the generated list at https://www.jenkins.io/doc/developer/extensions/credentials/#credentialsprovider ?

Reply:
That link is misleading as some of the external secrets manager plugins use a hybrid approach where secrets are stored in the service, but Jenkins stores credentials internally which are basically placeholders for the external credentials. Due to the various ways that credentials plugins can be made, it seems as though not everyone has used the same approach for this.